File: 2025/regional-andaluza/H/code/py/EstrategiaEstelar.py
from math import gcd

def leer_estado():
    N = int(input())
    f = []
    for _ in range(N):
        l = list(map(int, input().split()))
        f.append([[l[0], l[1], l[2]], [l[3], l[4], l[5]], [l[6], l[7], l[8]]])

    return f

def mcm(a, b):
    return a * b // gcd(a, b)

def obtener_ciclo_eje(posiciones):
    posiciones_iniciales = posiciones[:]
    velocidades = [0] * len(posiciones)
    pasos = 0

    while True:
        # Actualizar velocidades según las reglas
        for i in range(len(posiciones)):
            for j in range(i + 1, len(posiciones)):
                if posiciones[i] < posiciones[j]:
                    velocidades[i] += 1
                    velocidades[j] -= 1
                elif posiciones[i] > posiciones[j]:
                    velocidades[i] -= 1
                    velocidades[j] += 1

        # Actualizar posiciones
        for i in range(len(posiciones)):
            posiciones[i] += velocidades[i]
        pasos += 1
        if posiciones == posiciones_iniciales and all(v == 0 for v in velocidades):
            return pasos

def calcular_dias_para_repeticion(flotas):
    ciclos = []
    for eje in range(3):  # x, y, z
        posiciones = [flota[eje] for flota in flotas]
        ciclos.append(obtener_ciclo_eje(posiciones[:]))
    return mcm(ciclos[0], mcm(ciclos[1], ciclos[2]))

if __name__ == "__main__":
    flotas = leer_estado()
    for f in flotas:
        print(calcular_dias_para_repeticion(f))

File: 2025/regional-andaluza/H/code/cpp/EstrategiaEstelar.cpp
#include <iostream>
#include <vector>
#include <numeric> // std::gcd
using namespace std;

// Leer estado inicial: 3 líneas con 3 coordenadas (x, y, z)
vector<vector<int>> leer_estado() {
    vector<vector<int>> estado(3, vector<int>(3));
    for (int i = 0; i < 3; ++i)
        for (int j = 0; j < 3; ++j)
            cin >> estado[i][j];
    return estado;
}

int mcm(int a, int b) {
    return a * b / gcd(a, b);
}

int obtener_ciclo_eje(vector<int> posiciones) {
    vector<int> posiciones_iniciales = posiciones;
    vector<int> velocidades(3, 0);
    int pasos = 0;

    while (true) {
        // Actualizar velocidades
        for (int i = 0; i < 3; ++i) {
            for (int j = i + 1; j < 3; ++j) {
                if (posiciones[i] < posiciones[j]) {
                    velocidades[i]++;
                    velocidades[j]--;
                } else if (posiciones[i] > posiciones[j]) {
                    velocidades[i]--;
                    velocidades[j]++;
                }
            }
        }

        // Actualizar posiciones
        for (int i = 0; i < 3; ++i) {
            posiciones[i] += velocidades[i];
        }

        pasos++;

        if (posiciones == posiciones_iniciales && velocidades == vector<int>{0, 0, 0}) {
            return pasos;
        }
    }
}

int calcular_dias_para_repeticion(const vector<vector<int>>& flotas) {
    vector<int> ciclos;
    for (int eje = 0; eje < 3; ++eje) {
        vector<int> posiciones;
        for (int i = 0; i < 3; ++i) {
            posiciones.push_back(flotas[i][eje]);
        }
        ciclos.push_back(obtener_ciclo_eje(posiciones));
    }

    return mcm(ciclos[0], mcm(ciclos[1], ciclos[2]));
}

int main() {
    vector<vector<int>> flotas = leer_estado();
    cout << calcular_dias_para_repeticion(flotas) << endl;
    return 0;
}
